thriftbp/server.go

```diff
@@ -91,38 +96,232 @@ func NewServer(cfg ServerConfig) (*thrift.TSimpleServer, error) {
 	return server, nil
 }
 
+// NewServerFromOptions creates a new ServerConfig instance and delegates
+// to NewServer so that it can instantiate a new thrift.TSimpleServer.
+// This is a low level api and should only be used at the recommendation
+// of the baseplate maintainers. Most application implementations should
+// rely on NewBaseplateServer.
+func NewServerFromOptions(opts ...ServerOption) (*thrift.TSimpleServer, error) {
+	cfg, err := buildConfig(opts...)
+	if err != nil {
+		return nil, fmt.Errorf("failed to build server config: %w", err)
+	}
+	return NewServer(cfg)
+}
+
 // NewBaseplateServer returns a new Thrift implementation of a Baseplate
-// server with the given config.
+// server with the given config. This is the primary constructor that
+// should be used when building a baseplate server.
```

Comment:
similar style nit as above, in general we would prefer to limit the first paragraph of the doc comment to a single sentence (ends with the first `.`), and add a blank line to start the next one.